L1Trigger/TrackerTFP/test/AnalyzerKF.cc

```diff
@@ -0,0 +1,359 @@
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "SimTracker/TrackTriggerAssociation/interface/StubAssociation.h"
+#include "L1Trigger/TrackTrigger/interface/Setup.h"
+#include "L1Trigger/TrackerTFP/interface/DataFormats.h"
+#include "L1Trigger/TrackerTFP/interface/LayerEncoding.h"
+#include "L1Trigger/TrackerTFP/interface/KalmanFilterFormats.h"
+
+#include <TProfile.h>
+#include <TH1F.h>
+#include <TEfficiency.h>
+
+#include <vector>
+#include <deque>
+#include <set>
+#include <cmath>
+#include <numeric>
+#include <sstream>
+
+using namespace std;
+using namespace edm;
+using namespace tt;
+
+namespace trackerTFP {
+
+  /*! \class  trackerTFP::AnalyzerKF
+   *  \brief  Class to analyze hardware like structured TTTrack Collection generated by Kalman Filter
+   *  \author Thomas Schuh
+   *  \date   2020, Sep
+   */
+  class AnalyzerKF : public one::EDAnalyzer<one::WatchRuns, one::SharedResources> {
+  public:
+    AnalyzerKF(const ParameterSet& iConfig);
+    void beginJob() override {}
+    void beginRun(const Run& iEvent, const EventSetup& iSetup) override;
+    void analyze(const Event& iEvent, const EventSetup& iSetup) override;
+    void endRun(const Run& iEvent, const EventSetup& iSetup) override {}
+    void endJob() override;
+
+  private:
+    //
+    void associate(const TTTracks& ttTracks, const StubAssociation* ass, set<TPPtr>& tps, int& sum, const vector<TH1F*>& his, TProfile* prof) const;
+
+    // ED input token of accepted Tracks
+    EDGetTokenT<StreamsStub> edGetTokenAcceptedStubs_;
+    // ED input token of accepted Stubs
+    EDGetTokenT<StreamsTrack> edGetTokenAcceptedTracks_;
+    // ED input token of lost Stubs
+    EDGetTokenT<StreamsStub> edGetTokenLostStubs_;
+    // ED input token of lost Tracks
+    EDGetTokenT<StreamsTrack> edGetTokenLostTracks_;
+    // ED input token of TTStubRef to selected TPPtr association
```

Comment:
The word "selected" here is not very informative. One needs "selected for tracking efficiency". Similar comment in other analyzers.

Reply:
done